modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2ValueCacheObject.java

```diff
@@ -180,6 +183,22 @@ public Object getObject(boolean cpy) {
         return h1 > h2 ? 1 : -1;
     }
 
+    /**
+     * Stub GG-33962, GG-34893 (new comparator) workaround.
+     * Versions < 8.7.40 and < 8.8.11 are used incorrect comparator and a simple comparator change
+     * leads to the tree corruprion. Thus for "old" versions is used appropriate old comparator,
+     * we take into account that later all indexes will be rebuild and only one ver of comparator
+     * will be used.
+     */
+    public void useCorrectComparator() {
+        useFixedComparator = true;
```

Comment:
Using different names for the method and the field look a bit confusing.
Will the next be better? WDYT?
```suggestion
    public void useLagacyComparator() {
        useLagacyComparator= true;
```

Reply:
Still not clear.
Why do anyone want to use 'incorrect' comparator?